### PR TITLE
refactor: add region block identifiers

### PR DESCRIPTION
**Pull request by giulianoberteo**
## Summary of Pull Request

Replaced: `#### Start of section Name ####`

With: `#Region Section-Name`

For example:

```powershell
######### Start APIs for managing Application Virtual Networks ##########
######### End APIs for managing Application Virtual Networks ##########
```

is replaced with

```powershell
#Region AVN-Management
#EndRegion AVN-Management
```

## Type of Pull Request
- [ ] This is a bugfix.
- [ ] This is an enhancement or feature.
- [X] This is a code style / formatting update.
- [ ] This is a documentation update.
- [ ] This is a refactoring update.
- [ ] This is something else.
      Please describe:

## Context of the Pull Request
Using regions we can easily collapse/expand and group block of code instead of just looking at the commented sections, which can't be collapsed.

## Related to Existing Issues

Closes: #162

## Test and Documentation Coverage
- [X] Tests have been completed (for bugfixes / features).
- [ ] Documentation has been added / updated (for bugfixes / features).

## Breaking Changes?
- [ ] Yes, there are breaking changes.
- [X] No, there are no breaking changes.

### CODE REVIEW

**Review by tenthirtyam — CHANGES_REQUESTED** [review]
It appears that this pull request includes much more than implied in the summary.

For example, the `checkVCFToken` function( L3561) and `validateJsonInput` (L931)

**Ryan Johnson**
Senior Staff Solutions Architect | Product Engineering @ VMware, Inc.

**Comment by giulianoberteo** [comment]
@tenthirtyam fair enough, it was not intentional, let me review today.

File: PowerVCF.psm1 [hunk 1]
```diff
@@ -294,12 +294,10 @@ Function Add-VCFApplicationVirtualNetwork {
...
 Export-ModuleMember -Function Add-VCFApplicationVirtualNetwork
 
-######### End APIs for managing Application Virtual Networks ##########
-
-
+#EndRegion AVN-Management
 
-######### Start APIs for managing Backup and Restore ##########
 
+#Region BackupRestore-Management
```
Comment (GaryJBlake) [inline PowerVCF.psm1:300]: What about using the Region/EndRegion capability but leaving the headings as follows:

#Region APIs for managing Backup and Restore
.....
#EndRegion APIs for managing Backup and Restore

This achieves the same goal you were looking to add into the code file but maintains the descriptions as per the API guide.

Comment (giulianoberteo) [reply PowerVCF.psm1:300]: I was under the impression that it was not possible to use spaces on region names (from past uses).

That's why I came up with those suggested abbreviations. That said, I just tried and it seems to work just fine with a longer sentence such the one you've suggested. I'll update with a new commit and also fix the slipped validateJsonInput change (came from my other branch where I'm enhancing the function).

**Comment by giulianoberteo** [comment]
> For example, the `checkVCFToken` function (L3561)

@tenthirtyam what you're seeing there is that I moved the checkVCFToken inside the appropriate section. 
If you pay attention, in the current master branch the functions **checkVCFToken** and **Get-JWTDetail** are (wrongly) inside the section **APIs for managing Validations**
See L3590 to L3592 which is where the section **Start APIs for managing Access and Refresh Token** is but without any function grouped inside. 
Makes sense?

**Comment by giulianoberteo** [comment]
code fixed with new commit. all fixed as discussed, please review.

File: PowerVCF.psm1 [hunk 2]
```diff
@@ -939,13 +929,7 @@ Function Request-VCFCertificate {
...
         [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$domainName
     )
-
-    if (!(Test-Path $json)) {
-        Throw "JSON File Not Found"
-    }
-    else {
-        # Reads the requestCsrSpec json file contents into the $ConfigJson variable
-        $ConfigJson = (Get-Content -Raw $json)
+        validateJsonInput
```
Comment (tenthirtyam) [inline PowerVCF.psm1:932]: Slipped in?

File: PowerVCF.psm1 [hunk 3]
```diff
@@ -848,12 +844,7 @@ Function Request-VCFCertificateCSR {
         [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$json,
         [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$domainName
     )
-
-    if (!(Test-Path $json)) {
-        Throw "JSON File Not Found"
-    }
-    else {
-        $ConfigJson = (Get-Content -Raw $json) # Reads the requestCsrSpec json file contents into the $ConfigJson variable
+        validateJsonInput
```
Comment (tenthirtyam) [inline PowerVCF.psm1:847]: Slipped in? 🙃

**Review by tenthirtyam — CHANGES_REQUESTED** [review]
I see two more instances where `validateJsonInput` may have slipped into this pull request.

File: PowerVCF.psm1 (hunk 3 above)
Comment (giulianoberteo) [reply PowerVCF.psm1:847]: d'oh! damn it, yes, slipped yet again from the other branch. sorry, let me restore it to how it was

File: PowerVCF.psm1 (hunk 2 above)
Comment (giulianoberteo) [reply PowerVCF.psm1:932]: d'oh! damn it, yes, slipped yet again from the other branch. sorry, let me restore it to how it was

File: PowerVCF.psm1 (hunk 3 above)
Comment (giulianoberteo) [reply PowerVCF.psm1]: fixed

File: PowerVCF.psm1 (hunk 2 above)
Comment (giulianoberteo) [reply PowerVCF.psm1]: fixed

**Review by tenthirtyam — APPROVED** [review]
LGTM! 🚀

File: PowerVCF.psm1 [hunk 4]
```diff
@@ -2180,11 +2169,10 @@ Function Remove-VCFLicenseKey {
 }
 Export-ModuleMember -Function Remove-VCFLicenseKey
 
-######### End APIs for managing License Keys ##########
-
+#EndRegion APIs for managing License Keys
 
 
-######### Start APIs for managing Members of the Federation ##########
+#Region APIs for managing Members of the Federation
```
Comment (LifeOfBrianOC) [inline PowerVCF.psm1:2175]: Suggest merging this with #Region APIs for managing Federation. We should also consider deprecating these functions

Comment (giulianoberteo) [reply PowerVCF.psm1:2175]: I can regroup the functions no problem. 

How have you dealt with deprecating functions in the past ? put a banner to the user?

**Review by LifeOfBrianOC — APPROVED** [review]

File: PowerVCF.psm1 (hunk 4 above)
Comment (LifeOfBrianOC) [reply PowerVCF.psm1:2175]: I just checked and we already have a deprecation notice in the functions so i think we are good. Leaving the functions in place will avoid breaking backwards compatibility

"Write-Warning "Multi-Instance Management has been deprecated in VMware Cloud Foundation v4.4.0 and later, this API is no longer supported"

**Pull request merged**